src/yescrypt/yescrypt-platform.c

```diff
@@ -25,22 +25,76 @@
 #include <linux/mman.h> /* for MAP_HUGE_2MB */
 #endif
 
-#define HUGEPAGE_THRESHOLD		(12 * 1024 * 1024)
+/*
+ * JtR hack: on old systems like RHEL6, do use huge pages if available even at
+ * the risk of failing on munmap() if the kernel is reconfigured to use a
+ * non-default huge page size.  In upstream yescrypt, this tradeoff is resolved
+ * the other way around because failure of a critical system library like
+ * libxcrypt is deemed unacceptable.
+ */
+#ifndef MAP_HUGE_2MB
+#define MAP_HUGE_2MB 0
+#endif
```

Comment:
This is potentially very confusing but I guess it keeps the diff against upstream smaller?

Reply:
Yes, and it keeps not only the diff but also the code smaller. But I agree it's dirty, and not something I would do in upstream yescrypt.